ui/src/features/project/list/projects-list.tsx

```diff
@@ -1,53 +1,72 @@
 import { createQueryOptions, useQuery, useTransport } from '@connectrpc/connect-query';
 import { useQueries } from '@tanstack/react-query';
 import { Empty, Pagination } from 'antd';
-import { useState } from 'react';
 
 import { LoadingState } from '@ui/features/common';
 import {
   listProjects,
   listStages
 } from '@ui/gen/service/v1alpha1/service-KargoService_connectquery';
 
+import { useLocalStorage } from '../../../utils/use-local-storage';
+
 import { ProjectItem } from './project-item/project-item';
 import { ProjectListFilter } from './project-list-filter';
 import * as styles from './projects-list.module.less';
 
+const PAGE_SIZE_KEY = 'projects-page-size';
+const PAGE_NUMBER_KEY = 'projects-page-number';
+const FILTER_KEY = 'projects-filter';
+
 export const ProjectsList = () => {
-  const [pageSize, setPageSize] = useState(9);
-  const [page, setPage] = useState(1);
-  const [filter, setFilter] = useState('');
+  const [pageSize, setPageSize] = useLocalStorage(PAGE_SIZE_KEY, 10);
+  const [page, setPage] = useLocalStorage(PAGE_NUMBER_KEY, 1);
+  const [filter, setFilter] = useLocalStorage(FILTER_KEY, '');
 
   const { data, isLoading } = useQuery(listProjects, {
     pageSize: pageSize,
     page: page - 1,
     filter
   });
 
+  useEffect(() => {
```

Comment:
could you import this useEffect from React?

Reply:
my bad, imported it @Marvin9